src/wp-includes/class-wp-block.php

```diff
@@ -140,15 +140,47 @@ public function __construct( $block, $available_context = array(), $registry = n
 
 		$this->available_context = $available_context;
 
+		$this->refresh_context_dependents();
+		$this->refresh_parsed_block_dependents();
+	}
+
+	/**
+	 * Updates the context for the current block and its inner blocks.
+	 *
+	 * The method updates the context of inner blocks, if any, by passing down
+	 * any context values the block provides (`provides_context`).
+	 *
+	 * If the block has inner blocks, the method recursively processes them by creating new instances of `WP_Block`
+	 * for each inner block and updating their context based on the block's `provides_context` property.
+	 *
+	 * @since 6.8.0
+	 */
+	public function refresh_context_dependents() {
+		$this->available_context = array_merge( $this->available_context, $this->context );
+
 		if ( ! empty( $this->block_type->uses_context ) ) {
 			foreach ( $this->block_type->uses_context as $context_name ) {
 				if ( array_key_exists( $context_name, $this->available_context ) ) {
 					$this->context[ $context_name ] = $this->available_context[ $context_name ];
 				}
 			}
 		}
```

Comment:
I believe this method needs to call `refresh_parsed_block_dependents()` too?

That's because we modify the `$available_context` above, and that value affects what is passed to inner blocks.

In other words:
* If `$context` of an inner block is filtered, we need to call both methods.
* If `$parsed_block` of an inner block is filtered, we need to only call `refresh_parsed_block_dependents()`.